From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sun, 30 Dec 2018 09:26:12 +0100
Subject: Reintroduce override_build_timestamp

Revert "Remove override_build_date gn arg."

This reverts commit e335101eedd3ab28d16dcb4b22e5ee0b60044b89.
Use (overriden) build timestamp for chrome_version_id
---
 base/BUILD.gn             | 7 ++++++-
 build_overrides/build.gni | 5 +++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/base/BUILD.gn b/base/BUILD.gn
--- a/base/BUILD.gn
+++ b/base/BUILD.gn
@@ -3734,8 +3734,13 @@ action("build_date") {
 
   args = [
     rebase_path("$target_gen_dir/generated_build_date.h", root_build_dir),
-    build_timestamp,
   ]
+
+  if (override_build_timestamp != "N/A") {
+    args += [ override_build_timestamp ]
+  } else {
+    args += [ build_timestamp ]
+  }
 }
 
 if (enable_nocompile_tests) {
diff --git a/build_overrides/build.gni b/build_overrides/build.gni
--- a/build_overrides/build.gni
+++ b/build_overrides/build.gni
@@ -57,6 +57,11 @@ declare_args() {
   # due to 4GiB file size limit, see https://crbug.com/648948.
   # Set this flag to true to skip the assertion.
   ignore_elf32_limitations = false
+
+  # Override this value to give a specific build date.
+  # See //base/build_time.cc and //build/write_build_date_header.py for more
+  # details; expected format is Unix Epoch time in seconds.
+  override_build_timestamp = "N/A"
 }
 
 if (host_os == "mac" || is_apple) {
